addons/binding/org.openhab.binding.smhi/src/main/java/org/openhab/binding/smhi/internal/SmhiHandler.java

```diff
@@ -0,0 +1,440 @@
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.smhi.internal;
+
+import static org.openhab.binding.smhi.internal.SmhiBindingConstants.*;
+
+import java.text.DateFormat;
+import java.text.DecimalFormat;
+import java.text.DecimalFormatSymbols;
+import java.text.SimpleDateFormat;
+import java.util.ArrayList;
+import java.util.Calendar;
+import java.util.Date;
+import java.util.HashMap;
+import java.util.List;
+import java.util.concurrent.ScheduledFuture;
+import java.util.concurrent.TimeUnit;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.eclipse.smarthome.core.library.types.DateTimeType;
+import org.eclipse.smarthome.core.library.types.DecimalType;
+import org.eclipse.smarthome.core.library.types.StringType;
+import org.eclipse.smarthome.core.thing.Channel;
+import org.eclipse.smarthome.core.thing.ChannelUID;
+import org.eclipse.smarthome.core.thing.Thing;
+import org.eclipse.smarthome.core.thing.ThingStatus;
+import org.eclipse.smarthome.core.thing.ThingStatusDetail;
+import org.eclipse.smarthome.core.thing.binding.BaseThingHandler;
+import org.eclipse.smarthome.core.types.Command;
+import org.eclipse.smarthome.core.types.RefreshType;
+import org.eclipse.smarthome.core.types.State;
+import org.eclipse.smarthome.core.types.UnDefType;
+import org.eclipse.smarthome.io.net.http.HttpUtil;
+import org.openhab.binding.smhi.internal.model.SmhiData;
+import org.openhab.binding.smhi.internal.model.SmhiParameterTables;
+import org.openhab.binding.smhi.internal.model.SmhiParameters;
+import org.openhab.binding.smhi.internal.model.SmhiTimeSeries;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.gson.Gson;
+import com.google.gson.GsonBuilder;
+
+/**
+ * The {@link smhiHandler} is responsible for handling commands, which are
+ * sent to one of the channels.
+ *
+ * @author Michael Parment - Initial contribution
+ */
+@NonNullByDefault
+public class SmhiHandler extends BaseThingHandler {
+
+    private final Logger logger = LoggerFactory.getLogger(SmhiHandler.class);
+    private @Nullable ScheduledFuture<?> refreshJob;
+
+    protected static final String URL = "http://opendata-download-metfcst.smhi.se/api/category/pmp3g/version/2/geotype/point/lon/%s/lat/%s/data.json";
+
+    // Timeout for weather data requests.
+    private static final int SMHI_TIMEOUT = 5000;
+
+    private Gson gson = new GsonBuilder().create();
+
+    private String apiRequest = "";
+
+    @Nullable
+    private SmhiConfiguration config;
+
+    public SmhiHandler(Thing thing) {
+        super(thing);
+    }
+
+    private SmhiParameterTables pt = new SmhiParameterTables();
```

Comment:
please use meaningful variable names. if I understoof your code right, this table is static. it doesn't feel right to create several instances of that. I have to think about this a little bit.

Reply:
me to.

Reply:
see my comment below. move these tables to the binding constants and use a static initializer. You can then also read the data from a resource file.